### PR TITLE
Refresh Archery project in preparation for a new release.

### PR DESCRIPTION
**Pull request by non**
This project hasn't needed any updates for awhile, so I've let it be.

At this point, it seemed like it was time for an update, if only because of all the new versions. I ended up deciding to "modernize" the project based on my 2015-era sense of best Scala practices. So far this means:

 * ☑ Bump versions of Scala, SBT, ScalaTest, ScalaStyle, sbt-bintray
 * ☑ Use aggressive set of compiler flags
 * ☑ Use Travis
 * ☑ Use scoverage (and codecov.io)

After doing these, I was motivated to add a bunch of tests to improve our coverage. In the process I caught a minor bug, which helps justify this whole exercise.

Still on my list:

 * ☐ Use sbt-release plugin
 * ☑ Consider introducing scalastyle
 * ☑ Move files around to slightly more standard locations
 * ☑ Improve documentation
 * ☐ Consider using tut to create a tutorial

### CODE REVIEW

**Comment by non** [comment]
OK, I think this is good to go.

Anyone interested in signing-off on this?

/cc @softprops @chrislewis @joshmarcus

File: benchmark/src/main/scala/main.scala [hunk 1]
```diff
@@ -6,29 +6,29 @@ import scala.util.Random.{nextFloat, nextInt, nextGaussian}
...
-  }
-}
-
-object HybridTree {
-  def apply[A](entries: Entry[A]*): HybridTree[A] = {
-    val base = SpatialIndex(entries)(p => (p.geom.x, p.geom.y))
-    HybridTree(base, Set.empty[Entry[A]], RTree.empty[A])
-  }
-}
+// import geotrellis.Extent
```
Comment (softprops) [inline benchmark/src/main/scala/main.scala]: can these just be removed?

File: core/src/test/scala/rtree.scala [hunk 2]
```diff
@@ -2,7 +2,8 @@ package archery
 
 import scala.collection.mutable.ArrayBuffer
 import scala.math.{ceil, min, max}
-import scala.util.Random.nextFloat
+import scala.util.Random.{nextGaussian, nextInt}
```
Comment (softprops) [inline core/src/test/scala/rtree.scala]: not sure if you'd be interested but scala backported some things that are in newer versions of java for better performing  random generators under in multi threaded envs, see [this guy](https://github.com/scala/scala/blob/2.10.x/src/forkjoin/scala/concurrent/forkjoin/ThreadLocalRandom.java#L35)

File: benchmark/src/main/scala/main.scala (hunk 1 above)
Comment (non) [inline benchmark/src/main/scala/main.scala]: I am hoping to re-add them when I update to a newer version of GeoTrellis.

Comment (non) [inline benchmark/src/main/scala/main.scala]: But I could also delete them now and worry about it later.

File: core/src/test/scala/rtree.scala (hunk 2 above)
Comment (non) [inline core/src/test/scala/rtree.scala]: Oh interesting, thanks!

Comment (non) [inline core/src/test/scala/rtree.scala]: I will open an issue to investigate this later.

**Comment by non** [comment]
I think I addressed most of @softprops's coments, I'm going to go ahead and merge now.

**Pull request merged**